Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -3409,6 +3417,46 @@ - (void)_didRemoveLayerForFindOverlay
     _page->findClient().didRemoveLayerForFindOverlay(_page.get());
 }
 
+- (BOOL)_haveSetUnobscuredSafeAreaInsets
+{
+    return _haveSetUnobscuredSafeAreaInsets;
+}
+
+- (void)_resetUnobscuredSafeAreaInsets
+{
+    _haveSetUnobscuredSafeAreaInsets = NO;
+    _unobscuredSafeAreaInsets = { };
+    [self _scheduleVisibleContentRectUpdate];
+}
+
+- (BOOL)_hasOverriddenLayoutParameters
+{
+    return _viewLayoutSizeOverride
+        || _minimumUnobscuredSizeOverride
+        || _maximumUnobscuredSizeOverride;
+}
+
+- (CGSize)_viewLayoutSizeOverride
+{
+    return _viewLayoutSizeOverride.value_or(CGSizeMake(0, 0));
```

Comment:
Why are these getting flattened to something that is indistinguishable from a valid overridden value? Can you not just return an optional?

Reply:
I didn't want to add a C++ optional to an Objective-C header.

Reply:
But also, that's why there's a getter for whether any values are overridden.

Reply:
@hortont424 pointed out that there are already C++ concepts in that header; will fix.